core/partitioning/shape_analysis.cpp

```diff
@@ -1,168 +1,169 @@
-
-  if (jit_results_ivalues.isTuple()) {
-    auto results = jit_results_ivalues.toTuple()->elements();
-    for (auto r : results) {
-      jit_results.push_back(r);
-    }
-  } else {
-    jit_results.push_back(jit_results_ivalues);
-  }
-
-  size_t idx = 0;
-  for (auto& output : seg_block.raw_outputs()) {
-    ivalues_maps[output] = jit_results[idx++];
-  }
-
-  // set input shape for each segmented block so we wil use it in conversion process
-  std::vector<ir::Input> input_shapes;
-  std::vector<at::ScalarType> input_types;
-  for (auto& i : seg_block.raw_inputs()) {
-    if (ivalues_maps[i].isTensor()) {
-      // set the input_shape and data_type
-      // we can use a temp value here instead of replacing the values in ivalues_map since we only use ivalues_map for
-      // shape inference
-      auto cur_ivalue = ivalues_maps[i];
-      at::ScalarType t = cur_ivalue.toTensor().scalar_type();
-      if (!partition_info.truncate_long_and_double && (t == at::kLong || t == at::kDouble)) {
-        TORCHTRT_THROW_ERROR(
-            "Unable to process subgraph input type of at::kLong/at::kDouble, try to compile model with truncate_long_and_double enabled");
-      } else if (partition_info.truncate_long_and_double && t == at::kLong) {
-        cur_ivalue = cur_ivalue.toTensor().to(at::kInt);
-        LOG_WARNING("Truncating graph input type from at::kLong to at::kInt");
-      } else if (partition_info.truncate_long_and_double && t == at::kDouble) {
-        cur_ivalue = cur_ivalue.toTensor().to(at::kFloat);
-        LOG_WARNING("Truncating graph input type from at::kDouble to at::kFloat");
-      }
-      c10::optional<nvinfer1::DataType> dtype = util::optTypeMetaToTRTDataType(cur_ivalue.toTensor().dtype());
-      if (dtype == c10::nullopt) {
-        TORCHTRT_THROW_ERROR("Unsupported input data type " << cur_ivalue.toTensor().dtype());
-      }
-      if (cur_ivalue.toTensor().sizes().size() == 0) {
-        // handle Scalar types, which has sizes of []
-        input_shapes.push_back(util::toVec(util::toDims(c10::List<long int>({1}))));
-      } else {
-        input_shapes.push_back(util::toVec(util::toDims(cur_ivalue.toTensor().sizes())));
-      }
-      input_types.push_back(cur_ivalue.toTensor().scalar_type());
-    }
-  }
-
-  seg_block.register_inshapes(input_shapes);
-  seg_block.register_intypes(input_types);
-}
-
-void runShapeAnalysis(
-    std::vector<SegmentedBlock>& segmented_blocks,
-    std::unordered_map<const torch::jit::Value*, torch::jit::IValue>& example_tensor_map,
-    const PartitionInfo& partition_info) {
-  // register every segment's input shape, and it's running output IValues
-  for (auto& seg_block : segmented_blocks) {
-    torch::jit::ConstantPooling(seg_block.g());
-    getSegmentsOutputByRunning(seg_block, example_tensor_map, partition_info);
-  }
-  return;
-}
-
-} // namespace partitioning
-} // namespace core
-} // namespace torch_tensorrt
+#include "core/partitioning/shape_analysis.h"
```

Comment:
Any idea why this file is changed?

Reply:
Yes. I think the line endings across files in the repo are inconsistent. Apparently the line endings of that file changed when I committed.
Do you enforce a particular line endings? Which one should it be?

Reply:
We use UNIX line endings (LF)